### PR TITLE
chore: add Github workflow to build and test the driver

### PR DESCRIPTION
**Pull request by jwilliams-ocient**
The workflow is triggered on pull requests targeting `master` or `feature/*` branches.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -21,40 +23,51 @@ ARG METABASE_VERSION
...
+ARG MB_OCIENT_TEST_HOST
+ARG MB_OCIENT_TEST_PORT
+COPY test/ /build/metabase/modules/drivers/ocient/test
+COPY --from=stg_build /build/metabase/resources/modules/ocient.metabase-driver.jar /build/metabase/plugins/
+RUN clojure -X:test:deps prep
+# Some dependencies still get downloaded when the command below is run, but I'm not sure why
+
+ENV MB_OCIENT_TEST_HOST=${MB_OCIENT_TEST_HOST}
+ENV MB_OCIENT_TEST_PORT=${MB_OCIENT_TEST_PORT}
+ENV Drivers=ocient
```
Comment (benschwartz10) [inline Dockerfile:59]: Nit: Drivers -> DRIVERS

File: Dockerfile [hunk 2]
```diff
@@ -21,40 +23,51 @@ ARG METABASE_VERSION
...
+    bin/build-driver.sh ocient
+
+# Test stage
+FROM stg_build as stg_test
+ARG MB_OCIENT_TEST_HOST
+ARG MB_OCIENT_TEST_PORT
+COPY test/ /build/metabase/modules/drivers/ocient/test
+COPY --from=stg_build /build/metabase/resources/modules/ocient.metabase-driver.jar /build/metabase/plugins/
+RUN clojure -X:test:deps prep
+# Some dependencies still get downloaded when the command below is run, but I'm not sure why
```
Comment (benschwartz10) [inline Dockerfile:55]: IIRC the metabase code is set up to generate missing dependencies when needed. So when running the tests, it sounds like some might be missing. As long as it uses `ocient.metabase-driver.jar` and doesn't generate its own, I think this is fine.

File: test/metabase/driver/ocient_unit_test.clj [hunk 3]
```diff
@@ -0,0 +1,34 @@
+(ns metabase.driver.ocient-unit-test
```
Comment (benschwartz10) [inline test/metabase/driver/ocient_unit_test.clj:1]: Is there a reason for splitting up ocient-test and ocient-unit-test?

File: Dockerfile (hunk 1 above)
Comment (jwilliams-ocient) [reply Dockerfile:59]: Is this case-insensitive?

File: test/metabase/driver/ocient_unit_test.clj (hunk 3 above)
Comment (jwilliams-ocient) [reply test/metabase/driver/ocient_unit_test.clj:1]: the distinction is that unit tests do not require an Ocient database to run, whereas that is a requiremnt of the integration tests

File: Dockerfile (hunk 1 above)
Comment (benschwartz10) [reply Dockerfile:59]: I think it is, so it shouldn't matter. If it is case sensitive, [I believe the correct capitalization is DRIVERS](https://github.com/Xeograph/metabase/blob/master/test/metabase/test/data/datasets.clj#L10).

Comment (jwilliams-ocient) [reply Dockerfile:59]:
```suggestion
ENV DRIVERS=ocient
```

**Pull request merged**